### ISSUE TITLE
t/utf8.t - utf8_to_uvchr_buf failure on several systems

### ISSUE THREAD

**Issue by atoomic** [Dual-Life/Devel-PPPort#147]
notice the error on several version of Perl 5.20.3, 5.24.4, ... on FreeBSD
but also on some older Perl versions and linux, view cpantesters list below 

```
# Test 70 got: "7" (t/utf8.t at line 230 fail #7)
#    Expected: "13" (returned length utf8_to_uvchr_buf("\xff\x80\x90\x90\x90\xbf\xbf\xbf\xbf\xbf\xbf\xbf\xbf"); warnings disabled)
#  t/utf8.t line 230 is:         ok($ret->[1], $test->{'no_warnings_returned_length'},
```

_______________________________________________________________________________________________
Devel-PPPort-3.54:
- arm-freebsd / 5.20.3:
  - FAIL http://www.cpantesters.org/cpan/report/db87077a-f4b8-11e9-a65e-fe3acca03743 ("Nigel Horne" <njh@bandsman.co.uk ((root))>)

- arm-freebsd / 5.22.4:
  - FAIL http://www.cpantesters.org/cpan/report/1c2f616c-f4b6-11e9-a65e-fe3acca03743 ("Nigel Horne" <njh@bandsman.co.uk ((root))>)

- arm-freebsd / 5.24.4:
  - FAIL http://www.cpantesters.org/cpan/report/8bc6b6c4-f4b1-11e9-a65e-fe3acca03743 ("Nigel Horne" <njh@bandsman.co.uk ((root))>)

- arm-freebsd-multi / 5.20.3:
  - FAIL http://www.cpantesters.org/cpan/report/98b4ddb4-f4c2-11e9-a65e-fe3acca03743 ("Nigel Horne" <njh@bandsman.co.uk ((root))>)

- arm-freebsd-multi / 5.22.4:
  - FAIL http://www.cpantesters.org/cpan/report/05922ef2-f4b7-11e9-a65e-fe3acca03743 ("Nigel Horne" <njh@bandsman.co.uk ((root))>)

- arm-freebsd-multi / 5.24.4:
  - FAIL http://www.cpantesters.org/cpan/report/75544590-f4b2-11e9-a65e-fe3acca03743 ("Nigel Horne" <njh@bandsman.co.uk ((root))>)

- arm-freebsd-thread-multi / 5.20.3:
  - FAIL http://www.cpantesters.org/cpan/report/79cb4c98-f4c3-11e9-a65e-fe3acca03743 ("Nigel Horne" <njh@bandsman.co.uk ((root))>)

- arm-freebsd-thread-multi / 5.22.4:
  - FAIL http://www.cpantesters.org/cpan/report/f1be102a-f4b7-11e9-a65e-fe3acca03743 ("Nigel Horne" <njh@bandsman.co.uk ((root))>)

- arm-freebsd-thread-multi / 5.24.4:
  - FAIL http://www.cpantesters.org/cpan/report/5f7bb996-f4b3-11e9-a65e-fe3acca03743 ("Nigel Horne" <njh@bandsman.co.uk ((root))>)

- armv6l-linux-thread-multi / 5.16.1:
  - FAIL http://www.cpantesters.org/cpan/report/bf7ab350-f4c3-11e9-a65e-fe3acca03743 ("Nigel Horne" <njh@bandsman.co.uk ((root))>)

- i386-freebsd / 5.20.0:
  - FAIL http://www.cpantesters.org/cpan/report/6c815456-f53b-11e9-9066-e374b0ba08e8 ("Chris Williams (BINGOS)" <chris@bingosnet.co.uk>)

- i386-freebsd-thread-multi / 5.20.1:
  - FAIL http://www.cpantesters.org/cpan/report/09b60d16-f4c4-11e9-9066-e374b0ba08e8 ("Chris Williams (BINGOS)" <chris@bingosnet.co.uk>)

- i686-linux / 5.12.0:
  - FAIL http://www.cpantesters.org/cpan/report/3d4a3c56-f523-11e9-9066-e374b0ba08e8 ("Chris Williams (BINGOS)" <chris@bingosnet.co.uk>)

- i686-linux / 5.12.1:
  - FAIL http://www.cpantesters.org/cpan/report/3b1e27c0-f4d4-11e9-9066-e374b0ba08e8 ("Chris Williams (BINGOS)" <chris@bingosnet.co.uk>)

- i686-linux / 5.12.2:
  - FAIL http://www.cpantesters.org/cpan/report/74c8cbda-f478-11e9-9066-e374b0ba08e8 ("Chris Williams (BINGOS)" <chris@bingosnet.co.uk>)

- i686-linux-thread-multi / 5.12.1:
  - FAIL http://www.cpantesters.org/cpan/report/78f0e1d4-f504-11e9-9066-e374b0ba08e8 ("Chris Williams (BINGOS)" <chris@bingosnet.co.uk>)

- i686-linux-thread-multi / 5.12.2:
  - FAIL http://www.cpantesters.org/cpan/report/bfaf818e-f4a5-11e9-9066-e374b0ba08e8 ("Chris Williams (BINGOS)" <chris@bingosnet.co.uk>)
________________________________________________________________________________________________

copy: @khwilliamson

**Comment by khwilliamson** [comment]
On 10/23/19 9:18 AM, Nicolas R. wrote:
> notice the error on several version of Perl 5.20.3, 5.24.4, ... on FreeBSD
> but also on some older Perl versions and linux, view cpantesters list below
> 
> |# Test 70 got: "7" (t/utf8.t at line 230 fail #7) # Expected: "13" 
> (returned length 
> utf8_to_uvchr_buf("\xff\x80\x90\x90\x90\xbf\xbf\xbf\xbf\xbf\xbf\xbf\xbf"); 
> warnings disabled) # t/utf8.t line 230 is: ok($ret->[1], 
> $test->{'no_warnings_returned_length'}, |

I don't see  how this is related to UVCHR_SKIP()
That test is related to utf8_to_uvchr_buf.
Please explain.

**Comment by atoomic** [comment]
I read too fast and set an incorrect title but I've then corrected the
subject
you are right this is about utf8_to_uvchr_buf

**Comment by khwilliamson** [comment]
This should be fixed in 3.56
